c7n/filters/health.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2016 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import itertools
+from c7n.utils import local_session, chunks, type_schema
+from .core import Filter
+
+
+class healthEventFilter(Filter):
+    """Check if there are health events related to the resources"""
+
+    schema = type_schema(
+            'health-events',
+            eventTypeCodes={'type': 'array', 'items': {'type': 'string'}},
```

Comment:
these params should be case normalized and simplified. ie. `types` and `statuses` would be fine. also statuses should be an enum with the valid values ( 'open', 'closed', 'upcoming',)